src/Illuminate/Support/Traits/DispatchesActions.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Illuminate\Support\Traits;
+
+use Exception;
+use Illuminate\Auth\Access\AuthorizationException;
+use Illuminate\Container\Container;
+use Illuminate\Support\Attributes\UnauthorizedActionOverride;
+use ReflectionClass;
+use ReflectionNamedType;
+
+trait DispatchesActions
+{
+    public static function dispatch(mixed ...$params): mixed
+    {
+        $action = static::make(...$params);
+
+        if ($action->authorize()) {
+            return $action->handle();
+        }
+
+        $reflection = new ReflectionClass($action);
+        $attributes = $reflection->getAttributes(UnauthorizedActionOverride::class);
+
+        if (! empty($attributes)) {
+            return $attributes[0]->newInstance()->value;
+        }
+
+        throw new AuthorizationException('This action is unauthorized.');
+    }
+
+    public static function make(mixed ...$params): static
+    {
+        $reflector = new ReflectionClass(static::class);
+        $constructor = $reflector->getConstructor();
+
+        if (! $constructor) {
+            return new static;
+        }
+
+        $container = Container::getInstance();
+        $dependencies = [];
+        $paramsIndex = 0;
+
+        /**
+         * First attempt to resolve the service class from the container.
```

Comment:
I would use a comma here. This also improves consistency.
```suggestion
         * First, attempt to resolve the service class from the container.
```